ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -3677,11 +3677,14 @@ private final synchronized boolean doOnePhase ()
     {
         if (TxControl.onePhase)
         {
-            return (((pendingList == null) || (pendingList.size() == 1)) ? true
-                    : false);
+            if(pendingList == null) {
+                return true;
+            }
+            if (pendingList.size() == 1) {
+                return pendingList.peekFront().isPermittedTopLevelOnePhaseCommit();
```

Comment:
I expect we could mix records in one transaction, should not the whole list be checked?

Reply:
Sorry, pending list == 1, I see it now